src/System.Diagnostics.Process/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -67,16 +69,12 @@ internal static PerformanceCounterLib GetPerformanceCounterLib(string machineNam
             else
                 machineName = machineName.ToLowerInvariant();
 
-            LazyInitializer.EnsureInitialized(ref s_libraryTable, ref s_internalSyncObject, () => new Dictionary<string, PerformanceCounterLib>());
+            LazyInitializer.EnsureInitialized(ref s_libraryTable, ref s_internalSyncObject, () => new ConcurrentDictionary<string, PerformanceCounterLib>());
 
             string libraryKey = machineName + ":" + lcidString;
```

Comment:
It's unfortunate we're allocating a string just to serve as a key to the dictionary.  Either as a separate commit or a follow-up change, we could instead just make the key to the dictionary be a value tuple and avoid that allocation entirely.